packages/gatsby-theme-newrelic/src/components/InlineSignup.js

```diff
@@ -0,0 +1,203 @@
+  }
+
+  &:invalid:not(:focus-within):not(:placeholder-shown) + ${ValidationHint} {
+    visibility: visible;
+  }
+`;
+
+const InputContainer = styled.div`
+  display: grid;
+  gap: 0.25rem 0;
+  grid-template-rows: auto 1fr;
+  place-items: center;
+  position: relative;
+
+  &:first-of-type {
+    margin-left: 0.5rem;
+  }
+`;
+
+const Terms = styled.p`
+  color: var(--system-text-muted-light);
+  grid-column: 1 / 4;
+  font-size: 0.875rem;
+  line-height: 1.25;
+  margin-top: 0.25rem;
+
+  a {
+    color: currentColor;
+  }
+`;
+
+/**
+ * [VSU] This component allows users to sign up inline in a doc.
+ * It only renders if the current user is not logged in.
+ */
+const InlineSignup = () => {
+  const [error, setError] = useState(null);
+  const tessen = useTessen();
+  const { t } = useThemeTranslation();
+  const { loggedIn } = useLoggedIn();
+  const submitEvent = {
+    category: 'InlineSignup',
+    eventName: 'attemptedSignup',
+  };
+
+  const onSubmit = async (e) => {
+    e.preventDefault();
+    const { email, name } = Object.fromEntries(new FormData(e.target));
+    const organizationId = await createAccountRequest(
+      { email, name },
+      tessen,
+      submitEvent
+    );
+
+    if (organizationId) {
+      const redirectUrl = `https://newrelic.com/thank-you?org=${organizationId}&standalone=true`;
+      window.location.assign(redirectUrl);
+    } else {
+      setError(true);
+    }
+  };
+
+  if (loggedIn == null || loggedIn) return null;
+
+  return (
+    <Form onSubmit={onSubmit}>
+      <CTAText>{t('inlineSignup.ctaText')}</CTAText>
+
+      <InputContainer>
+        <label className="screenreader-only" htmlFor="inline-signup-name">
+          {t('inlineSignup.nameLabel')}
+        </label>
+        <Input
+          className="first"
+          id="inline-signup-name"
+          name="name"
+          // this basically says that a name has to have atleast one letter.
+          // numbers are allowed, for a name like Charles the 3rd.
+          // i used `\p{Letter}` here instead of `[a-zA-Z]`
+          // to allow Unicode characters in names, like Björk, or 준영.
+          pattern=".*\p{Letter}+.*"
+          placeholder={t('inlineSignup.nameLabel')}
+          required
+          type="text"
+        />
+        {/* for the sake of not complicating the markup, i'm using this
+         * for both the input validation hint and the error text if
+         * the account creation fails.
+         */}
+        <ValidationHint className={cx({ error })}>
+          {error
+            ? t('inlineSignup.submitError')
+            : t('inlineSignup.nameValidationHint')}
+        </ValidationHint>
+      </InputContainer>
+      <InputContainer>
+        <label className="screenreader-only" htmlFor="inline-signup-email">
+          {t('inlineSignup.emailLabel')}
+        </label>
+        <Input
+          className="last"
+          id="inline-signup-email"
+          name="email"
+          pattern=".+@.+\..+"
```

Comment:
the validation logic for the Feedback form doesn't require users to list a top level domain for the email.  We went with this to emulate the signup form on newrelic.com/signup

Reply:
well nvm, it does required a top level domain now, so we should update the feedback form. diff PR

Reply:
oh interesting. i'm kind of surprised it ever allowed an email without a TLD 😮. i can update the validation in Feedback in a different change

Reply:
opened #870 for this